bigframes/session/__init__.py

```diff
@@ -793,19 +797,26 @@ def _read_gbq_table(
         # Create Ibis table expression and validate
         # -----------------------------------------
 
+        if table_ref.dataset_id.startswith("_"):
+            time_travel_timestamp = None
+
         # Use a time travel to make sure the DataFrame is deterministic, even
         # if the underlying table changes.
         # TODO(b/340540991): If a dry run query fails with time travel but
         # succeeds without it, omit the time travel clause and raise a warning
         # about potential non-determinism if the underlying tables are modified.
-        table_expression = bf_read_gbq_table.get_ibis_time_travel_table(
-            ibis_client=self.ibis_client,
-            table_ref=table_ref,
+        sql = bf_io_bigquery.to_query(
+            f"{table_ref.project}.{table_ref.dataset_id}.{table_ref.table_id}",
             index_cols=index_cols,
             columns=columns,
-            filters=filters,
+            sql_predicate=bf_io_bigquery.compile_filters(filters) if filters else None,
             time_travel_timestamp=time_travel_timestamp,
+            # If we've made it this far, we know we don't have any
+            # max_results to worry about, because in that case we will
+            # have executed a query with a LIMIT clause.
+            max_results=None,
         )
+        bf_read_gbq_table.validate_sql_through_ibis(sql, self.ibis_client)
```

Comment:
We'll need Henry's logic to dryrun with and without the `time_travel_timestamp` so we can continue to support tables that don't support time travel.